inspirehep/modules/records/texkeys.py

```diff
@@ -0,0 +1,332 @@
+from inspirehep.utils.helpers import force_list
+from inspirehep.utils.record import get_value
+
+IS_PYTHON_3 = sys.version_info[0] == 3
+
+
+def generate_texkey(record):
+    """It generates the texkey in according to the data inside the record.
+
+    :param record: the record that needs the texkey
+    :type record: dict
+
+    :return: return the new texkey if it is
+    able to produce it, otherwise empty string
+    :rtype: string
+    """
+    new_texkey = ''
+    first_part = first_part_generator(record)
+    if first_part:
+        new_texkey = first_part + ':' + second_part_generator(record) + third_part_generator()
+    return new_texkey
+
+
+def first_part_generator(record):
+    """It generates the first part of the texkey.
+
+    :param record: the record that needs the texkey
+    :type record: dict
+
+    :return: return the first part of the texkey
+    if it is able to produce it, otherwise empty string
+    :rtype: string
+    """
+    if 1 <= len(record.get('authors', [])) < 10:
+        first_part = sanitize(record.get('authors')[0]['full_name'])
+    elif record.get('collaborations', []):
+        first_part = sanitize(record.get('collaborations')[0]['value'])
+    elif record.get('corporate_author', []):
+        first_part = sanitize(record.get('corporate_author')[0])
+    elif 'proceedings' in record.get('document_type', []):
+        first_part = 'proceedings'
+    elif record.get('authors', []):
+        first_part = sanitize(record.get('authors')[0]['full_name'])
+    else:
+        first_part = ''
+
+    return first_part
+
+
+def second_part_generator(record):
+    """It generates the second part of the texkey.
+
+    :param record: the record that needs the texkey
+    :type record: dict
+
+    :return: return the year of the record if it
+    is able to extract it, otherwise the current year
+    :rtype: string
+    """
+    def _get_year_from_record(record):
+        earliest_date = extract_earliest_date(record)
+
+        if not earliest_date:
+            return str(date.today().year)
+
+        return re.search(r'\d{4}', earliest_date).group(0)
+
+    return _get_year_from_record(record)
+
+
+def third_part_generator():
+    """It generates the second part of the texkey.
+
+    :param record: the record that needs the texkey
+    :type record: dict
+
+    :return: return the year of the record if it
+    is able to extract it, otherwise the current year
+    :rtype: string
+    """
+    random_sequence = get_three_random_letters()
+    return random_sequence
+
+
+def is_current_texkey_valid(record):
+    """It checks if inside the record there is a valid texkey.
+
+    :param record: the record that needs the texkey
+    :type record: dict
+
+    :return: return True if there is a valid texkey otherwise False
+    :rtype: boolean
+    """
+    fixed_part = first_part_generator(record) + ':' + second_part_generator(record)
+    previous_texkeys = record.get('texkeys')
```

Comment:
If you do `record.get('texkeys', [''])`, you can get rid of the subsequent `if`.